src/ProductDetails.js

```diff
@@ -0,0 +1,104 @@
+import styled from "styled-components";
+import { useParams } from "react-router-dom";
+import { useProductById } from "./utils/hooks/useProductById";
+import { useProductCategories } from "./utils/hooks/useProductCategories";
+import Slideshow from "./Slider/Slideshow";
+
+const ShopcartButton = styled.button`
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+  padding: 5px 20px 5px 20px;
+  background-color: rgba(0, 150, 255, 0.8);
+  font-size: 20px;
+  font-weight: bold;
+  text-decoration: none;
+  color: black;
+  border-radius: 15px;
+
+  &:hover {
+    background-color: rgba(0, 150, 255, 0.2);
+  }
+`;
+
+const ProductLabel = styled.label`
+  display: block;
+`;
+
+const ProductDetails = () => {
+  const params = useParams();
+
+  const { data: categoriesData, isLoading: isLoadingCategories } =
+    useProductCategories();
+
+  const {
+    data: { results },
+    isLoading: isLoadingProduct,
+  } = useProductById(params.productId);
+
+  let productSlides = [];
+  let product = null;
+
+  let category = "";
+
+  if (isLoadingProduct === false) {
+    let [{ data: productData }] = results;
+
+    product = productData;
```

Comment:
Optaría por mejor hacer un spread de este objeto en lugar de solo asignarlo. Así 
```suggestion
    product = { ...productData };
```
La razón de esto tiene que ver con unos conceptos más profundos de JavaScript, lo que estás haciendo actualmente es solo agregar una nueva forma de **ACCEDER** al `productData`, lo que sugiero yo, es para **CREAR** un objeto nuevo a partir de los valores del mismo, que es la intención original de esta línea.

Ahora ya lo siguiente lo dejaría como opcional, pero personalmente creo que la variable `product` no tiene razón de ser, solo es un tercero que recibe exactamente el mismo contenido que `productData`, así que yo la retornaría tal cual como viene de results, eso si, cambiando su nombre de `productData` a `product`, como originalmente planeas usarla en el componente.

Reply:
Lo que pasa es que la variable product no la pude destructurar directamente en la llamada al custom hook... creo que es porque cuando se hace la llamada inicial, el hook devuelve un objeto vacio y **results** es indefinido... y cuando intentaba destructurar algo de results, me marcaba errores de que "el objeto no era iterable" o algo por el estilo. La unica manera que encontre de eliminar el error fue hacer la destructuracion hasta que el objeto estuviera listo. Igual y hay una manera mas elegante de solucionarlo pero fue lo que pude hacer jeje

Reply:
Refactorice partes de este componente en mi nuevo entregable pero aun asi si tuve que seguir destructurando esa variable aparte... si tienes una sugerencia sobre eso lo agradeceria